addons/website_sale/static/src/js/website_sale.js

```diff
@@ -738,6 +749,37 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
         }
         this._applyHash();
     },
+    /**
+     * @private
+     */
+    async _getCurrentWebsite() {
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: (context) => {
+                websiteId = context['website_id'];
+            },
+        });
+        return this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [[websiteId], ['cart_add_on_page']],
```

Comment:
It should be done on xml side, to avoid to make 1 rpc at each click on add to cart no?

Reply:
Actually, I added the rpc call to get the updated value of "cart_add_on_page".
By adding attribute in xml like this: `t-att-data-stay-on-page="not request.website.is_public_user() and request.website.cart_add_on_page"`, the option will keep one value until page reloaded (even if it was changed in settings).

But since it's not very critical to have an updated value of the option on every click, I think we can do this in xml or just move the rpc call to `init` method and set the value in `this.stayOnPageOption`. what do you think?